generatebundlefile/promote.go

```diff
@@ -297,13 +302,30 @@ func (c *SDKClients) moveImage(image Image, authFile string) error {
 
 	err := copyImage(BundleLog, authFile, sourcePrefix+":"+image.Tag, destinationPrefix+":"+image.Tag)
 	if err != nil {
-		return fmt.Errorf("unable to copy image tag from source to destination repo %w", err)
+		return fmt.Errorf("unable to copy image tag from source to destination repo: %w", err)
 	}
 
 	// when the image.Tag doesn't match image.Digest in the source ECR, we need to move the Digest directly
 	err = copyImage(BundleLog, authFile, sourcePrefix+"@"+image.Digest, destinationPrefix+"@"+image.Digest)
 	if err != nil {
-		return fmt.Errorf("unable to copy image digest from source to destination repo %w", err)
+		return fmt.Errorf("unable to copy image digest from source to destination repo: %w", err)
+	}
+
+	if regionalBuildMode {
+		destinationPrefix = fmt.Sprintf("docker://%s/%s", c.ecrPublicClient.SourceRegistry, image.Repository)
+		BundleLog.Info(fmt.Sprintf("Promoting images to secondary ECR destination: %s/%s", c.ecrPublicClient.SourceRegistry, image.Repository))
+
+		err := copyImage(BundleLog, authFile, sourcePrefix+":"+image.Tag, destinationPrefix+":"+image.Tag)
```

Comment:
Q: how is it guaranteed that only the dependencies of curated-packages controller moved to the public registry? Looks like copyImage doesn't do this check.

Reply:
We do that check [here](https://github.com/aws/eks-anywhere-packages/blob/main/generatebundlefile/hack/promote.sh#L37)